### ISSUE TITLE
[macOS] NIX_SSL_CERT_FILE: unbound variable

### ISSUE THREAD

**Issue by bflyblue** [NixOS/nix#2193]
I believe this patch has introduced an error when installing on OS X without 
NIX_SSL_CERT_FILE being set in the environment:

https://github.com/NixOS/nix/commit/c4b9486f9b20d319a7ac3ac2553bdc0fa63c3899

```
---- sudo execution ------------------------------------------------------------
I am executing:

    $ sudo HOME=/var/root /nix/store/2jp9d9qns3372hkx4k523hc1bbzkr4yp-nix-2.0.3/bin/nix-env -i /nix/store/2jp9d9qns3372hkx4k523hc1bbzkr4yp-nix-2.0.3

to installing a bootstrapping Nix in to the default Profile

installing 'nix-2.0.3'
building '/nix/store/jp5371ld47218v5bzc6m5584l8b8w2n1-user-environment.drv'...
created 7 symlinks in user environment
/var/folders/4t/sfj5mbdn7693nqhj99vz0k800000gn/T/nix-binary-tarball-unpack.XXXXXXXXXX.uw14ghw7/unpack/nix-2.0.3-x86_64-darwin/install-multi-user: line 730: NIX_SSL_CERT_FILE: unbound variable

---- oh no! --------------------------------------------------------------------
```

I was able to work around with `export NIX_SSL_CERT_FILE=""` before running `install-nix`.

**Comment by bflyblue** [comment]
Thanks for sorting it out so quickly! I know Nix on OS X doesn't get as many eyes so I'm glad to help :)

**Comment by grahamc** [comment]
Actually ... it is worse than it seems. The PR breaking this on macOS was written for a deployment to macOS hosts. I evidently failed to test the case where NIX_SSL_CERT_FILE isn't set before running the installer. I thought I had. In fact, I thought I had tested it this morning -- not a case of not trying, but I _thought_ the variable was unset! I don't know how I made that mistake.

Thank you for reporting it, I will try to not ever let this happen again.
